gcloud-java-core/src/main/java/com/google/gcloud/BasePage.java

```diff
@@ -67,10 +67,10 @@ public int hashCode() {
 
   @Override
   public boolean equals(Object obj) {
-    if (!(obj instanceof BaseListResult)) {
+    if (!(obj instanceof BasePage)) {
```

Comment:
This is another reason why I would like to see only one Page implementation.
Especially to avoid `basePage.equals(specificPage)` but `!specificePage.equals(basePage)`